internal/service/quicksight/data_set.go

```diff
@@ -0,0 +1,2726 @@
+							Required: true,
+						},
+						"custom_sql": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							MaxItems:     1,
+							ExactlyOneOf: []string{"custom_sql", "relational_table", "s3_source"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data_source_arn": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: verify.ValidARN,
+									},
+									"name": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringLenBetween(1, 64),
+									},
+									"sql_query": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringLenBetween(1, 65536),
+									},
+									"columns": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MinItems: 1,
+										MaxItems: 2048,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"name": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringLenBetween(1, 128),
+												},
+												"type": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringInSlice(quicksight.InputColumnDataType_Values(), false),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"relational_table": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							MaxItems:     1,
+							ExactlyOneOf: []string{"custom_sql", "relational_table", "s3_source"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data_source_arn": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: verify.ValidARN,
+									},
+									"input_columns": {
+										Type:     schema.TypeList,
+										Required: true,
+										MinItems: 1,
+										MaxItems: 2048,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"name": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringLenBetween(1, 128),
+												},
+												"type": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringInSlice(quicksight.InputColumnDataType_Values(), false),
+												},
+											},
+										},
+									},
+									"name": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringLenBetween(1, 64),
+									},
+									"catalog": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringLenBetween(0, 256),
+									},
+
+									"schema": {
```

Comment:
"schema" should have a ValidateFunc to ensure it is no longer than 64 characters ([API reference](https://docs.aws.amazon.com/quicksight/latest/APIReference/API_RelationalTable.html#:~:text=Required%3A%20No-,Schema,Required%3A%20No,-See%20Also))